nexus/src/db/datastore.rs

```diff
@@ -4330,9 +4337,14 @@ impl DataStore {
             .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
     }
 
-    /// Look up a granted device authorization token.
-    // TODO-security: authz
-    pub async fn device_auth_get_token(
+    /// Look up a granted device access token.
+    /// Note: since this lookup is not by a primary key or name,
```

Comment:
Is the client allowed to fetch the token more than once?

Wondering out loud if there should be another table of short-lived records here whose primary key _is_ (client_id, device_id), which just has a foreign key into the token table.  If they're only ever allowed to fetch the token once, that might also be useful to enforce that.  If this doesn't seem appealing, ignore.

Reply:
A good question. I don't _think_ there's an explicit restriction on fetching the token more than once, though I do think it would be odd. I'm not very keen to add a new table, though.